docs/working_with_data_sources.md

```diff
@@ -0,0 +1,222 @@
+
+# Create Amazon Redshift data source with access key and secret key authentication
+redshift_iam_auth_model = {
+            'dataSource' => {
+                'name' => 'RedshiftDS_IAM',
+                'alias' => GoodData::Rest::Connection.generate_string,
+                'prefix' => 'prefix_',
+                'connectionInfo' => {
+                    'redshift' => {
+                        'url' => 'jdbc:redshift://test',
+                        'authentication' => {
+                            'iam' => {
+                                'dbUser' => 'user_iam_test',
+                                'accessKeyId' => 'OWIUEWQIO723628',
+                                'secretAccessKey' => 'abc123'
+                            }
+                        },
+                        'database' => 'database_test',
+                        'schema' => 'schema_test'
+                    }
+                }
+            }
+        }
+client = GoodData.connect
+redshift_iam_auth_ds = client.create(GoodData::DataSource, redshift_iam_auth_model)
+redshift_iam_auth_ds.save
+```
+
+Or you can create a Google BigQuery data source:
+
+```ruby
+# encoding: utf-8
+
+require 'gooddata'
+
+bigquery_ds_model = {
+          'dataSource' => {
+              'name' => 'My Data Source',
+              'alias' => 'ds_alias',
+              'prefix' => 'ds_prefix_',
+              'connectionInfo' => {
+                  'bigQuery' => {
+                      'authentication' => {
+                          'serviceAccount' => {
+                              'clientEmail' => 'test@abc.com',
+                              'privateKey' => '***'
+                          }
+                      },
+                      'project' => 'project_test',
+                      'schema' => 'schema_test'
+                  }
+              }
+          }
+      }
+client = GoodData.connect
+bigquery_ds = client.create(GoodData::DataSource, bigquery_ds_model)
+bigquery_ds.save
+```
+
+Currently the platform also supports data sources as Amazon S3 data source, Generic data source, Agile Data Warehouse Service data source ... 
+
+Renaming Data Source
+-------
+
+If you need to rename the data source:
+
+```ruby
+require 'gooddata'
+
+GoodData.connect
+ds = GoodData::DataSource.from_id('datasource_id')
+ds.name = "New data source name"
+ds.save
+```
+
+Get data source from specific identify or alias
+-------
+
+You can get data source from specify data source identify or alias:
+
+```ruby
+# encoding: utf-8
+
+require 'gooddata'
+
+GoodData.connect
+ds_from_id = GoodData::DataSource.from_id('datasource_id')
+ds_from_alias = GoodData::DataSource.from_alias('datasource_alias')
+```
+
+Get all data sources
+-------
+
+You can get all data sources from Platform:
+
+
+```ruby
+# encoding: utf-8
+
+require 'gooddata'
+
+GoodData.connect
+ds_list = GoodData::DataSource.all
+```
+
+Delete a Data Source
+-------
+
+You can delete a data source with specific identify or alias:
```

Comment:
You can delete this line, it's clear from the heading what this code sample is going to be about.